pkg/util/sliding_window.go

```diff
@@ -0,0 +1,218 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package util
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+// PollingFunc is polled at specified intervals to collect the value
+// that should be included in the slidingWindow data
+type PollingFunc func() (value float64)
+
+// CallbackFunc is invoked when the polling function collects data and
+// we need to use the newly-calculated values
+type CallbackFunc func(SlidingWindow)
```

Comment:
This comment is pretty confusing.  Is it called when the stats are updated?  Maybe `StatsUpdatedFunc`?